File: smart-society-frontend/src/services/residentService.js
import API from "../api/axiosConfig";

export const getAllResidents = () => {
  return API.get("/residents");
};

export const getResidentById = (id) => {
  return API.get(`/residents/${id}`);
};

export const addResident = (data) => {
  return API.post("/residents", data);
};

export const updateResident = (id, data) => {
  return API.put(`/residents/${id}`, data);
};

export const deleteResident = (id) => {
  return API.delete(`/residents/${id}`);
};

File: SmartSociety/src/main/java/com/example/SmartSociety/service/ResidentService.java
package com.example.SmartSociety.service;


import com.example.SmartSociety.entity.Resident;

import java.util.List;

public interface ResidentService {

    Resident addResident(Resident resident);

    List<Resident> getAllResidents();

    Resident getResidentById(Long id);

    Resident updateResident(Long id, Resident resident);

    void deleteResident(Long id);
}
